src/app/dataset-view/additional-components/metadata-component/components/set-transform/components/queries-section/queries-section.component.ts

```diff
@@ -0,0 +1,78 @@
+import {
+    ChangeDetectionStrategy,
+    Component,
+    Input,
+    OnChanges,
+    SimpleChanges,
+} from "@angular/core";
+import { SqlQueryStep } from "src/app/api/kamu.graphql.interface";
+import * as monaco from "monaco-editor";
+import { ViewportScroller } from "@angular/common";
+
+@Component({
+    selector: "app-queries-section",
+    templateUrl: "./queries-section.component.html",
+    styleUrls: ["./queries-section.component.sass"],
+    changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class QueriesSectionComponent implements OnChanges {
+    @Input() public queries: Omit<SqlQueryStep, "__typename">[];
+    @Input() public datasetName: string;
+    public readonly sqlEditorOptions: monaco.editor.IStandaloneEditorConstructionOptions =
+        {
+            theme: "vs",
+            language: "sql",
+            renderLineHighlight: "none",
+            minimap: {
+                enabled: false,
+            },
+        };
+
+    constructor(private scroll: ViewportScroller) {}
+
+    ngOnChanges(changes: SimpleChanges): void {
+        const currentQueries = changes.queries.currentValue as Omit<
+            SqlQueryStep,
+            "__typename"
+        >[];
+        if (!currentQueries.length) {
+            this.queries.push({
+                alias: this.datasetName,
+                query: "",
+            });
+        }
+    }
+
+    public isLastQuery(index: number): boolean {
+        return index === this.queries.length - 1;
+    }
+
+    public isPenultimateQuery(index: number): boolean {
```

Comment:
There was a game series called Tropico, about tropical dictators managing their island. The dictator's assistant was named Penulltimo, meaning 1st after the 1st. :-)

Reply:
;)Then will be isBeforeLastQuery()